i18n infrastructure + first end-user facing pages translations

ghys opened the pull request:
Related to #394.

Add vue-i18n.
The localized messages are in `@/assets/i18n/{domain}/{locale}.json`,
and are localized by component.
They should be compatible with Crowdin.
They are also referenced in single-file components with this trick:
```
import { loadLocaleMessages } from '@/js/i18n'

export default {
  ...
  i18n: {
    messages: loadLocaleMessages(require.context('@/assets/i18n/{domain}'))
  }
```
That way they should be code splitted away from the entry point bundle
when their related component is, and they are scoped to that component.

The `common` domain with a few common and home page related messages
is loaded on initialization.

English and French messages are provided for the pages accessible to end users.
German (to be proofed!) is provided only for the `common` domain.

A footer message is added to the settings & developer tools menu if
the locale has common messages to inform the user that these parts are
not translated yet.

Signed-off-by: Yannick Schaus <github@schaus.net>

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "home.cards.analyzeAll": "Alles analysieren",
+  "home.otherApps": "Andere Apps",
+  "home.editHome": "Homepage bearbeiten",
+  "home.pinToHome": "Pin zum Startbildschirm",
```
Should be „An Startbildschirm anheften“

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "home.editHome": "Homepage bearbeiten",
+  "home.pinToHome": "Pin zum Startbildschirm",
+  "home.exitToApp": "Zurück zur App",
+  "home.tip.otherApps": "Dieses Panel öffnen, um andere Webinterfaces zu starten",
```
Andere Anwendungen

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "sidebar.noPages": "Keine Seiten",
+  "sidebar.administration": "Verwaltung",
+  "sidebar.settings": "Einstellungen",
+  "sidebar.developerTools": "Entwicklung",
```
Entwickler Tools

ghys replied:
Nice, I had tried "Entwicklerwerkzeuge" but it didn't fit in the android theme... This works better.

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "sidebar.settings": "Einstellungen",
+  "sidebar.developerTools": "Entwicklung",
+  "sidebar.helpAbout": "Über & Hilfe",
+  "sidebar.unlockAdmin": "Verwaltung entsperren",
```
Administration entsperren

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "sidebar.tip.signIn": "Anmelden um Zugriff auf die Einstellungswerkzeuge",
+  "page.navbar.back": "@:dialogs.back",
+  "page.navbar.edit": "Bearbeiten",
+  "admin.notTranslatedYet": "Der Verwaltungsbereich ist derzeit noch nicht übersetzt."
```
Der Administrationsbereich ist noch nicht übersetzt

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/common/de.json:
```diff
@@ -0,0 +1,41 @@
+  "sidebar.developerTools": "Entwicklung",
+  "sidebar.helpAbout": "Über & Hilfe",
+  "sidebar.unlockAdmin": "Verwaltung entsperren",
+  "sidebar.tip.signIn": "Anmelden um Zugriff auf die Einstellungswerkzeuge",
```
Anmelden für Administration

hmerk commented on bundles/org.openhab.ui/web/src/assets/i18n/empty-states/de.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "overview.title": "Das Übersichtsseite is noch nicht konfiguriert",
```
Die Übersichtsseite....

stritti replied:
`Die Übersichtsseite ist noch nicht konfiguriert`

hmerk commented:
@ghys I left some comments on behalf of the german translation.
Rest should be ok.

ghys commented:
Thanks!
If you want you can translate `about/en.json`, `theme-switcher/en.json` and `profile/en.json` 😉

hmerk commented:
If I find some time during my tomorrows booring meetings, I will do so....

hmerk commented:
profile/german
  "profile.title": "Dein Profil",
   "profile.changePassword": "Paswort ändern",
   "profile.sessions": "Sitzungen",
   "profile.sessions.header": "Achtung, eine gelöschte Sitzung kann aktiv bleiben, bis alle gültigen Zugriffstokens abgelaufen sind.",
   "profile.sessions.signOut": "Von aktiver Sitzung abmelden",
   "profile.sessions.created": "Erstellt: ",
   "profile.sessions.lastRefreshed": "Letzte Aktualisierung: ",
   "profile.apiTokens": "API Tokens",
   "profile.apiTokens.header": " Erstelle API Tokens um externen Anwendungen und Diensten ohne OAuth2 Unterstützung dauerhaften Zugriff zu gewähren.",
   "profile.apiTokens.created": "Erstellt: ",
   "profile.apiTokens.validForScope": "Gültig für Anwendung: ",
   "profile.apiTokens.create": "Erzeuge neues API Token“

hmerk commented:
about/german
  "about.title": "Über",
   "about.homePage": "Startbildschirm",
   "about.documentation": "Dokumentation",
   "about.communityForum": "Community Forum",
   "about.technicalInformation": "Technische Information",
   "about.technicalInformation.configurationFolder": "Konfigurationsverzeichnis",
   "about.technicalInformation.userdataFolder": "Verzeichnis für Anwenderdaten",
   "about.technicalInformation.logsFolder": "Verzeichnis für Protokolldateien",
   "about.technicalInformation.operatingSystem": "Betriebssystem",
   "about.technicalInformation.javaRuntime": "Java Runtime",
   "about.technicalInformation.resourceStats": "{nbproc} Verfügbare Prozessorleistung - {ram} Verfügbarer Arbeitsspeicher",
   "about.technicalInformation.viewDetails": "Details",
   "about.appearanceOptions": "Erscheinungsbild (nur für dieses Gerät)",
   "about.reload": "Neu laden",
   "about.reload.purgeExplanation1": "Zwischenspeicher und/oder Service Worker werden zur lokalen Speicherung von Resourcen verwendet, um die Ladezeiten zu verkürzen. Eine zuverlässige Erkennung eines Updates auf eine neue Version kann nicht sichergestellt werden.",
   "about.reload.purgeExplanation2": " Wähle unten Zwischenspeicher löschen und aktualisieren um alle Zwischenspeicher zu löschen, Service Worker abzumelden und alle Informationen erneut vom Server zu laden.",
   "about.reload.purgeCachesAndRefresh": "Zwischenspeicher löschen und aktualisieren",
   "about.reload.reloadApp": "Anwendung neu laden"

hmerk commented:
theme-switcher/german
  "about.theme": "Thema",
   "about.theme.auto": "Automatisch",
   "about.darkMode": "Dunkelmodus",
   "about.darkMode.auto": "Automatisch",
   "about.darkMode.light": "Hell",
   "about.darkMode.dark": "Dunkel",
   "about.navigationBarsStyle": "Navigationsleisten Sil",
   "about.miscellaneous": "Verschiedenes",
   "about.miscellaneous.home.navbar": "Einfache Navigationsleiste auf dem Startbildschirm",
   "about.miscellaneous.home.background": "Standardfarbe für den Startbildschirm",
   "about.miscellaneous.home.hideChatInput": "Chat Eingabebox auf dem Startbildschirm nicht anzeigen",
   "about.miscellaneous.home.disableCardExpansionAnimation": "Animation für Kartenerweiterung abschalten",
   "about.miscellaneous.theme.disablePageTransition": "Animation für Seitenübergänge abschalten"
 }

hmerk commented:
@ghys German translation finished,see comments above... 😉

stritti commented:
There is a typo:
> profile/german
> "profile.changePassword": "Paswort ändern",
"profile.changePassword": "Passwort ändern",

BTW, is there already a Crowdin-Integration?

ghys commented:
I saw the typo yesterday, fixed in https://github.com/openhab/openhab-webui/pull/602/commits/10dc573e3007e1b6de2d5aed68eed043394c7919 :) Also "about.navigationBarsStyle": "Navigationsleisten Sil" -> Stil

Also @hmerk missed a few messages I added in the meantime:
"about.reload.confirmPurge": "Purge all application caches and unregister the service workers? This will also reload the page from the server, which might take a few seconds.",
"profile.sessions.delete.success": "Session deleted",
  "profile.sessions.delete.error": "Error while deleting the session: ",
  "profile.sessions.signOut.error": "Error while signing out: ",
"profile.apiTokens.create": "Create new API token",
  "profile.apiTokens.delete.success": "API token deleted",
  "profile.apiTokens.delete.error": "Error while deleting the API token: "

> BTW, is there already a Crowdin-Integration?

No, but I'll try to coordinate with @kaikreuzer to have it today or tomorrow. I'd like to throw in the analyzer window messages and the setup wizard, and then I believe that'll be enough for the release. There shouldn't be any untranslated messages left shown to users who don't have admin access, I believe, which was the goal stated in https://github.com/openhab/openhab-webui/issues/394#issuecomment-739545492.

hmerk commented:
@ghys Sorry, was too late yesterday, here are the missing translations: 

"about.reload.confirmPurge": "Lösche den Zwischenspeicher und melde die Service Worker ab, Dies führt zum Neuladen der Seite und dauert einige Sekunden.",
"profile.sessions.delete.success": "Sitzung gelöscht",
"profile.sessions.delete.error": "Fehler beimLöschen der Sitzung: ",
"profile.sessions.signOut.error": "Fehler bei der Abmeldung: ",
"profile.apiTokens.create": "Erstelle neuen API Token",
"profile.apiTokens.delete.success": "API Token gelöscht",
"profile.apiTokens.delete.error": "Fehler beim Löschen des API Token: "

hmerk commented:
Addition : 

theme-switcher/german
"about.theme": "Erscheinungsbild",

ghys commented:
Also "about.homePage": "Startbildschirm", > is that correct? It's not the home page as in page of the app, but the home page of the website: 
![image](https://user-images.githubusercontent.com/2004147/101455531-08e0d600-3933-11eb-8e05-7bf8122aa136.png)

Maybe I should just put "Website" there, in fact.

hmerk commented:
Correct, my fault, just name it "Internetseite"....

ghys commented:
new strings with no German ;)

```json
{
  "setupwizard.language": "Language",
  "setupwizard.region": "Region",
  "setupwizard.timezone": "Time Zone",
  "setupwizard.beginSetup": "Begin Setup",
  "setupwizard.skipSetup": "Skip Setup",
  "setupwizard.skipSetup.confirm.title": "Skip Setup",
  "setupwizard.skipSetup.confirm.message": "Are you sure? Setup saves you time by performing just a few basic configuration tasks. You should only skip it if you know what you're doing.",
  "setupwizard.location.title": "Set your Location",
  "setupwizard.location.header1": "Would you like to share your home's location?",
  "setupwizard.location.header2": "It will help determining data dependent on your position, like sunrise/sunset times or the weather.",
  "setupwizard.location.parameterLabel": "Location of your home",
  "setupwizard.location.retrieveFromDevice": "Retrieve from This Device",
  "setupwizard.location.retrieveFromDevice.error": "Error while retrieving current position",
  "setupwizard.location.retrieveFromDevice.notAvailable.title": "Sorry",
  "setupwizard.location.retrieveFromDevice.notAvailable.message": "Geolocation is not available",
  "setupwizard.location.footer": "This will ask your device for the permission to use its current location, only to help you fill in your current latitude and longitude above. You can revoke the permission afterwards.",
  "setupwizard.location.setLocation": "Set Location",
  "setupwizard.location.configureLater": "Configure in Settings Later",
  "setupwizard.addons.title": "Install Add-ons",
  "setupwizard.addons.header1": "Most of openHAB's functionality is provided by add-ons.",
  "setupwizard.addons.header2": "Choose which ones you'd like to install right away.",
  "setupwizard.addons.browseAddonsOnWebsite": "Browse Add-ons on openhab.org",
  "setupwizard.addons.selectAddons": "Select Add-ons to Install",
  "setupwizard.addons.selectAddons.placeholder": "Try: astro, mqtt, hue, knx...",
  "setupwizard.addons.footer": "To optimize your system resources, install only the add-ons you need! Installing add-ons can take a while. Please be patient and stay on this page until the operation finishes.",
  "setupwizard.addons.installAddons": "Install 1 add-on | Install {n} add-ons",
  "setupwizard.addons.installLater": "Install Add-ons Later",
  "setupwizard.addons.installing": "Installing Add-ons...",
  "setupwizard.addons.installingAddon": "Installing Add-on: {addon}",
  "setupwizard.addons.progress": "{current} of {total}",
  "setupwizard.addons.pleaseWait": "Please Wait...",
  "setupwizard.addons.waitMessage": "It may take a few minutes to install the add-ons you selected.",
  "setupwizard.welcome.title": "Welcome to openHAB!",
  "setupwizard.welcome.getStarted": "Get Started"
}
```

hmerk commented:
setupwizard/german
  "setupwizard.language": "Sprache",
  "setupwizard.region": "Region",
  "setupwizard.timezone": "Zeitzone",
  "setupwizard.beginSetup": "Starte Setup",
  "setupwizard.skipSetup": "Überspringe Setup",
  "setupwizard.skipSetup.confirm.title": "Überspringe Setup",
  "setupwizard.skipSetup.confirm.message": " Bust Du sicher ? Der Setupassistent spart Dir Zeit indem einige Grundlegende Einstellungen gemacht werden. Du solltest nur Überspringen wählen, wenn Du Dir absolut sicher bist.",
  "setupwizard.location.title": "Eigener Standort",
  "setupwizard.location.header1": "Möchtest Du Deinen eigenen Standort teilen?",
  "setupwizard.location.header2": "Standort teilen ist hilfreich bei der Ermittlung positionsabhängiger Informationen wie Sonnenauf-/untergang oder Wetterdaten.",
  "setupwizard.location.parameterLabel": "Eigener Standort",
  "setupwizard.location.retrieveFromDevice": "Ermittlung über Dein Gerät",
  "setupwizard.location.retrieveFromDevice.error": "Fehler bei der Positionsbestimmung",
  "setupwizard.location.retrieveFromDevice.notAvailable.title": "Entschuldigung",
  "setupwizard.location.retrieveFromDevice.notAvailable.message": "Geoposition ist nicht verfügbar",
  "setupwizard.location.footer": "Erlaubnis erteilen, über die Position Deines Gerätes die aktuellen Geo-Coordinaten (latitude/longitude) zu übernehmen. Diese Erlaubnis kann später widerrufen werden.",
  "setupwizard.location.setLocation": "Position speichern",
  "setupwizard.location.configureLater": "Später in den Einstellungen konfigurieren",
  "setupwizard.addons.title": "Installiere Add-ons",
  "setupwizard.addons.header1": "Die meisten Funktionalitäten von openHAB werden über Add-ons bereitgestellt.",
  "setupwizard.addons.header2": "Wähle, welche Add-ons Du jetzt installieren möchtest.",
  "setupwizard.addons.browseAddonsOnWebsite": "Informiere Dich über verfügbare Add-ons auf openhab.org",
  "setupwizard.addons.selectAddons": "Wähle zu installierende Add-ons",
  "setupwizard.addons.selectAddons.placeholder": "Versuche: astro, mqtt, hue, knx...",
  "setupwizard.addons.footer": "Zur optimalen Ausnutzung der Systemressourcen installiere bitte nur die Add-ons, die Du wirklich benötigst! Die Installation kann ein wenig Zeit in Anspruch nehmen. Bitte einen Moment Geduld und bleibe auf dieser Seite, bis die Anstallation abgeschlossen ist.",
  "setupwizard.addons.installAddons": "Installiere 1 add-on | Installiere {n} add-ons",
  "setupwizard.addons.installLater": "Installiere Add-ons später",
  "setupwizard.addons.installing": "Installiere Add-ons...",
  "setupwizard.addons.installingAddon": "Installiere Add-on: {addon}",
  "setupwizard.addons.progress": "{current} von {total}",
  "setupwizard.addons.pleaseWait": "Bitte warten...",
  "setupwizard.addons.waitMessage": "Es kann einen Moment dauern, bis alle ausgewählten Add-ons installiert sind.",
  "setupwizard.welcome.title": "Wilkommen bei openHAB!",
  "setupwizard.welcome.getStarted": "Beginne"

hmerk commented:
@ghys ... translated ...

ghys commented:
Thanks!

"setupwizard.skipSetup.confirm.message": " Bust Du sicher ? > I'm gonna assume it's "Bist"
"setupwizard.location.configureLater": "Später in den Einstellungen konfigurieren" > probably too long in the button, I'll use "Später konfigurieren"
"setupwizard.addons.installLater": "Installiere Add-ons später" > probably best to use "Später installieren" to be consistent with the other button?

ghys commented:
Looking good!

![2020-12-08_14-24-45](https://user-images.githubusercontent.com/2004147/101489499-51af8380-3961-11eb-8eae-4ed216f84aaa.gif)

hmerk commented:
@ghys Yes, Typo, "Bist" is correct....
I am fine with your two suggested shortenings....

stritti commented:
"setupwizard.skipSetup.confirm.message": " Bist Du sicher ?" > remove the blank bevor "?" -> "setupwizard.skipSetup.confirm.message": " Bist Du sicher?"

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/about/de.json:
```diff
@@ -0,0 +1,21 @@
+  "about.appearanceOptions": "Erscheinungsbild (nur für dieses Gerät)",
+  "about.reload": "Neu laden",
+  "about.reload.purgeExplanation1": "Zwischenspeicher und/oder Service Worker werden zur lokalen Speicherung von Resourcen verwendet, um die Ladezeiten zu verkürzen. Eine zuverlässige Erkennung eines Updates auf eine neue Version kann nicht sichergestellt werden.",
+  "about.reload.purgeExplanation2": " Wähle unten Zwischenspeicher löschen und aktualisieren um alle Zwischenspeicher zu löschen, Service Worker abzumelden und alle Informationen erneut vom Server zu laden.",
```
> "about.reload.purgeExplanation2": " Wähle unten Zwischenspeicher löschen und aktualisieren um alle [...]

about.reload.purgeExplanation2": " Wähle unten Zwischenspeicher löschen und aktualisieren, um alle [...]

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/empty-states/de.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "overview.title": "Die Übersichtsseite is noch nicht konfiguriert",
```
> "overview.title": "Die Übersichtsseite is noch nicht konfiguriert",

"overview.title": "Die Übersichtsseite ist [...]

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/profile/de.json:
```diff
@@ -0,0 +1,19 @@
+  "profile.title": "Dein Profil",
+  "profile.changePassword": "Passwort ändern",
+  "profile.sessions": "Sitzungen",
+  "profile.sessions.header": "Achtung, eine gelöschte Sitzung kann aktiv bleiben, bis alle gültigen Zugriffstokens abgelaufen sind.",
```
> Zugriffstokens 

Zugriffstoken

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/profile/de.json:
```diff
@@ -0,0 +1,19 @@
+  "profile.sessions.created": "Erstellt: ",
+  "profile.sessions.lastRefreshed": "Letzte Aktualisierung: ",
+  "profile.sessions.delete.success": "Sitzung gelöscht",
+  "profile.sessions.delete.error": "Fehler beimLöschen der Sitzung: ",
```
>   "profile.sessions.delete.error": "Fehler beimLöschen der Sitzung: ",

"profile.sessions.delete.error": "Fehler beim Löschen der Sitzung: ",

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/profile/de.json:
```diff
@@ -0,0 +1,19 @@
+  "profile.sessions.lastRefreshed": "Letzte Aktualisierung: ",
+  "profile.sessions.delete.success": "Sitzung gelöscht",
+  "profile.sessions.delete.error": "Fehler beimLöschen der Sitzung: ",
+  "profile.sessions.signOut.error": "Fehler bei der Abmeldung: ",
```
> "profile.sessions.signOut.error": "Fehler bei der Abmeldung: ",

"profile.sessions.signOut.error": "Fehler beim Abmelden: ",

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/semantics/de.json:
```diff
@@ -0,0 +1,94 @@
+{
+  "Temperature": "Temperaturen",
+  "Light": "Lichtleuchtung",
```
> "Light": "Lichtleuchtung",

Either:

"Light": "Licht",

or

"Light": "Beleuchtung",

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/semantics/de.json:
```diff
@@ -0,0 +1,94 @@
+  "OpenState": "Offener Zustand",
+  "Tampered": "Manipuliert",
+  "Tilt": "Neigung",
+  "Equipment": "Sonstiges",
```
> "Equipment": "Sonstiges",

"Equipment": "Gerät",

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/de.json:
```diff
@@ -0,0 +1,36 @@
+  "setupwizard.beginSetup": "Starte Setup",
+  "setupwizard.skipSetup": "Überspringe Setup",
+  "setupwizard.skipSetup.confirm.title": "Überspringe Setup",
+  "setupwizard.skipSetup.confirm.message": "Bist Du sicher ? Der Setupassistent spart Dir Zeit indem einige Grundlegende Einstellungen gemacht werden. Du solltest nur Überspringen wählen, wenn Du Dir absolut sicher bist.",
```
Remove blank bevor "?"

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/theme-switcher/de.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "about.theme": "Erscheinungsbild",
+  "about.theme.auto": "Auto",
```
Automatisch

ghys replied:
Doesn't work, it's for the little square box, I updated to "Auto." with a punkt to signify it's an abbreviation.
You'll be able to debate about this on Crowdin later ;)

stritti replied:
Okay, I see, not that big thing. Let me know when Crowdin is on start. I already support makecode for micro:bot ;)

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/theme-switcher/de.json:
```diff
@@ -0,0 +1,15 @@
+  "about.theme": "Erscheinungsbild",
+  "about.theme.auto": "Auto",
+  "about.darkMode": "Dunkelmodus",
+  "about.darkMode.auto": "Auto",
```
Automatisch

stritti commented on bundles/org.openhab.ui/web/src/assets/i18n/theme-switcher/de.json:
```diff
@@ -0,0 +1,15 @@
+  "about.miscellaneous": "Verschiedenes",
+  "about.miscellaneous.home.navbar": "Einfache Navigationsleiste auf dem Startbildschirm",
+  "about.miscellaneous.home.background": "Standardfarbe für den Startbildschirm",
+  "about.miscellaneous.home.hideChatInput": "Chat Eingabebox auf dem Startbildschirm nicht anzeigen",
```
For Usability I would suggest to invert this flag to be not negative ...

stritti commented:
BTW: For better translations I can recommend DeepL https://www.deepl.com/translator

ghys commented:
Ok, I believe this is in good shape. Thank you @hmerk and @stritti for the help in bootstrapping the German translations. The rest will likely (hopefully!) be done on Crowdin.

hmerk commented:
@ghys I have completed the german translations on crowdin, how will those be incorporated ???

ghys commented:
See https://community.openhab.org/t/help-translating-openhab-3/109802 😉
